File: SEGUNDO/PDOO/PRACTICAS/PRACTICA1/ShieldBooster.rb
#encoding:utf-8

module Deepspace
	class ShieldBooster
		def initialize (n, b, u) #"constructor"
				@name=n
				@boost=b
				@uses=u
		end
		
		def self.newCopy (copy)
			new(copy.name, copy.boost, copy.uses)
		end
		
		public
		def name
			@name
		end
		def boost
			@boost
		end
		def uses
			@uses
		end
		
		def useIt
			if @uses > 0
				@uses = @uses - 1
				@boost
			else
				1.0
			end
		end
	end
end

File: SEGUNDO/PDOO/PRACTICAS/PRACTICA2/java/deepSpace/src/deepspace/ShieldBooster.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package deepspace;

/**
 *
 * @author lauralazaro
 */
public class ShieldBooster {
    String name;
    float boost;
    int uses;
    
    ShieldBooster(String n, float b, int u) {
        name = n;
        boost = b;
        uses = u;
    }
    
    ShieldBooster (ShieldBooster s) {
        name = s.name;
        boost = s.boost;
        uses = s.uses;
    }
    
    ShieldToUI getUIversion () {
        return (new ShieldToUI(this));
    }
    
    float getBoost () {
        return boost;
    }
    
    int getUses () {
        return uses;
    }
    
    float useIt () {
        if (uses>0) {
            uses = uses-1;
            return boost;
        }
        else {
            return 1.0f;
        }
    }
    
    public String toString () {
        return getUIversion().toString();
    }
}
